### PR TITLE
initial commit of a functioning decoder for WSFM proxy data

### LINKED ISSUE #1511: weather satellite follow on microwave image IODA converter
## Description

create a decoder for the Weather Satellite Follow-on (WSF) Microwave Imager (MI)

The data is presented in 3 formats:
Native resolution
Resampled (B-G resampled CFOVs)
Resampled Earth Gridded

The resampled backus-gilbert will be the initial target. Here the sensor data record (SDR) will be decoded to obtain the brightness temperature data.

## Requirements

python3 interface with netCDF4 and H5PY, as well as the python bindings to the IODA converter library

## Acceptance Criteria (Definition of Done)

generation of a IODA file from the input orbital segments

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

This adds a functioning decoder for the WSF-M proxy data converting to IODA format

## Issue(s) addressed

Resolves #1511 

## Impact

Provides a WSF-M decoding capability

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

File: src/hdf5/wsfm_2ioda.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+float_missing_value = iconv.get_default_fill_val(np.float32)
+int_missing_value = iconv.get_default_fill_val(np.int32)
+long_missing_value = iconv.get_default_fill_val(np.int64)
+
+metaDataName = iconv.MetaDataName()
+obsValName = iconv.OvalName()
+
+# globals
+WSF_WMO_sat_ID = 999     # unknown
```
Comment (BenjaminRuston) [inline src/hdf5/wsfm_2ioda.py:38]: @nancylbaker think there is one assigned need to ask Steve

File: src/hdf5/wsfm_2ioda.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        ('longitude', metaDataName): [],
+        ('dateTime', metaDataName): [],
+        ('sensorScanPosition', metaDataName): [],
+        ('solarZenithAngle', metaDataName): [],
+        ('solarAzimuthAngle', metaDataName): [],
+        ('sensorZenithAngle', metaDataName): [],
+        ('sensorAzimuthAngle', metaDataName): [],
+        ('sensorViewAngle', metaDataName): [],
+    }
+#       ('satelliteAscendingFlag', metaDataName): [],
```
Comment (BenjaminRuston) [inline src/hdf5/wsfm_2ioda.py:269]: and need to also ask Steve if there is an ascending or descending flag in the data

**Review by BenjaminTJohnson — APPROVED** [review]
I have no way of testing this, but nothing reads as unusual.

File: src/hdf5/wsfm_2ioda.py [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+    obs_data[('satelliteIdentifier', metaDataName)] = np.full((nlocs), WMO_sat_ID, dtype='int32')
+
+    obs_data[('dateTime', metaDataName)] = np.array(np.repeat(f['ScanStartTime'][:] - et_offset, nbeam_pos), dtype='int64')
+
+    k = 'brightnessTemperature'
+    # have to reorder the channel axis to be last then merge ( nscans x nspots = nlocs )
+    # 'TbLowRes' - 'Brightness temperatures on the resampled scan grid for each polarimetric channel, 30 km RFOV'
+    # 'TbHighRes' - 'Brightness temperatures on the resampled scan grid for each polarimetric channel, 20 km RFOV'
+    # for channels less than 20 GHz use the TbLowRes and the TbHighRes for the other bands
+    freq_cf = 20.
```
Comment (BenjaminRuston) [inline src/hdf5/wsfm_2ioda.py:172]: @nancylbaker here is the `crossover` frequency `freq_cf` values below this use the `TbLowRes` and frequencies above use the `TbHighRes`

let's discuss the way forward with this as I think it is largely ready to go and we can make smaller changes from there (assuming this is the type of file we want to decode)

**Comment by BenjaminRuston** [comment]
@fcvdb please go ahead and try this converter out

**Comment by fcvdb** [comment]
The converter runs fine on Narwhal and Nautilus.

**Review by fcvdb — APPROVED** [review]
The converter runs fine on Narwhal and Nautilus.

**Comment by BenjaminRuston** [comment]
@nancylbaker thinking to just move forward with this and update as needed as new files are presented from WSFM

**Comment by nancylbaker** [comment]
Thanks for developing the prototype @BenjaminRuston and for testing this @fcvdb .
Agree with merging for now.

**Review by nancylbaker — APPROVED** [review]

**Pull request merged**